docs/configuration.rst

```diff
@@ -4,37 +4,161 @@
+                 environment that your project will run in,   and project-\
+                 e.g. the Scrapy version that will be used).  specific
+``image``        Whether to use a custom Docker image on      global default
+                 deploy. See :ref:`deploy-custom-image`.      and project-\
+                                                              specific
+``version``      Version tag to use when deploying. This can  global only
+                 be an arbitrary string or one of the magic   
+                 keywords ``AUTO`` (default), ``GIT``, or
+                 ``HG``. By default, ``shub`` will
+                 auto-detect your version control system and
+                 use its branch/commit ID as version.
+``apikey``       API key to use for deployments. You will     global only
+                 typically not have to touch this setting as
+                 it will be configured inside
+                 ``~/.scrapinghub.yml`` in your home
+                 directory, via ``shub login``.
+================ ============================================ =================
 
-    # Use git branch/commit as version when deploying. Other possible values
-    # are AUTO (default) or HG
-    version: GIT
-
-``projects`` is a mapping from human-friendly names to project IDs. This allows
-you to run ``shub deploy prod`` instead of ``shub deploy 33333``. ``version``
-is a string to be used as project version information when deploying to Scrapy
-Cloud. By default, ``shub`` will auto-detect your version control system and
-use its branch/commit ID as version.
-
-Typically, you will configure only your API keys and the version specifier in
-the global ``~/.scrapinghub.yml``, and keep project configuration in the local
-``project_dir/scrapinghub.yml``. However, sometimes it can be convenient to
-also configure projects in the global configuration file. This allows you to
-use the human-friendly project identifiers outside of the project directory.
+
+.. _`Scrapy Cloud stack`: http://help.scrapinghub.com/scrapy-cloud/scrapy-cloud-stacks
+
+
+Example configurations
+----------------------
+
+Custom requirements file and fixed version information::
+
+    project: 12345
+    requirements:
+      file: requirements_scrapinghub.txt
+    version: 0.9.9
+
+Custom Scrapy Cloud stack, requirements file and and additional private
+dependencies::
+
+    project: 12345
+    stack: scrapy:1.1
+    requirements:
+      file: requirements.txt
+      eggs:
+        - privatelib.egg
+        - path/to/otherlib.egg
+
+Using the latest Scrapy 1.3 stack in staging and development, but pinning the
+production stack to a specific release::
+
+    projects:
+      default: 12345
+      staging: 33333
+      prod:
+        id: 44444
+        stack: scrapy:1.3-py3-20170322
+ 
+    stack: scrapy:1.3-py3
+
+Using a custom Docker image::
+
+    projects:
+      default: 12345
+      prod: 33333
+
+    image: true
```

Comment:
Could you add an example with global `image: true` and per-project `image: false` - to show how to deploy stack if image is enabled globally?